--- CMakeLists.txt.orig	2018-06-29 13:25:52.538441303 +1000
+++ CMakeLists.txt	2018-06-29 14:15:14.038545896 +1000
@@ -29,9 +29,6 @@
   #SET(CMAKE_BUILD_TYPE "MINSIZEREL")
 ENDIF()
 
-set(CMAKE_CXX_STANDARD 98)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-
 set (PROJECT_BINARY_DIR bin)
 set (PROJECT_LIBRARY_DIR lib)
 set (PROJECT_SOURCE_DIR src)
@@ -147,6 +144,8 @@
     add_definitions(-D_CRT_SECURE_NO_WARNINGS)
     add_definitions(-D_CRT_NONSTDC_NO_WARNING)
     add_definitions(-D_SCL_SECURE_NO_WARNINGS)
+
+    add_definitions(-DH5_BUILT_AS_DYNAMIC_LIB)
     
     # by default the compiler produces gratuitous warnings. Disable some of them
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4290 /wd4005 /wd4101 /wd4244 /wd4800 /wd4251 /wd4996")
