diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 40302c2..6937f1c 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -25,7 +25,12 @@ project(extrat LANGUAGES CXX)
 find_package(Python 3.7 COMPONENTS Interpreter Development)
 
 # Find version of awkward we have installed
-EXECUTE_PROCESS(COMMAND ${Python_EXECUTABLE} -c "import awkward; print(awkward.__version__)" OUTPUT_VARIABLE AWKWARD_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+message(STATUS "Finding awkward version via Python")
+# don't import all of awkward - has some C libraries that won't work for cross compilation
+EXECUTE_PROCESS(COMMAND ${Python_EXECUTABLE} -c "from awkward._version import __version__;print(__version__)" OUTPUT_VARIABLE AWKWARD_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+if(AWKWARD_VERSION STREQUAL "")
+    message(FATAL_ERROR "Unable to import awkward")
+endif()
 message(STATUS "Using awkward version ${AWKWARD_VERSION}")
 
 # get awkward headers
