--- python/CMakeLists.txt.orig	2025-04-22 10:44:15.441738075 +1000
+++ python/CMakeLists.txt	2025-04-22 10:46:26.324608973 +1000
@@ -78,7 +78,7 @@
 target_link_libraries(testseg_libkea PRIVATE libkea::Kealib)
 
 # custom install step to create .pyc files
-install(CODE "execute_process(COMMAND ${Python_EXECUTABLE} -m compileall ${PYKEA_INSTALL_DIR})")
+#install(CODE "execute_process(COMMAND ${Python_EXECUTABLE} -m compileall ${PYKEA_INSTALL_DIR})")
 
 enable_testing()
 add_test(NAME testpython COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/testpython.py)
