--- python/CMakeLists.txt.orig	2025-04-22 10:44:15.441738075 +1000
+++ python/CMakeLists.txt	2025-04-22 12:51:51.269701839 +1000
@@ -78,7 +78,11 @@
 target_link_libraries(testseg_libkea PRIVATE libkea::Kealib)
 
 # custom install step to create .pyc files
-install(CODE "execute_process(COMMAND ${Python_EXECUTABLE} -m compileall ${PYKEA_INSTALL_DIR})")
+set(Python_EXECUTABLE_NATIVE "")
+cmake_path(NATIVE_PATH "${Python_EXECUTABLE}" Python_EXECUTABLE_NATIVE)
+message(STATUS "execute_process(COMMAND ${Python_EXECUTABLE_NATIVE} -m compileall ${PYKEA_INSTALL_DIR})")
+install(CODE "execute_process(COMMAND ${Python_EXECUTABLE_NATIVE} -m compileall ${PYKEA_INSTALL_DIR})")
+
 
 enable_testing()
 add_test(NAME testpython COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/testpython.py)
